### PR TITLE
Feat: fix Berlin hard fork test and transactions validation

### PR DESCRIPTION
**Pull request by mrLSD**
## Description

Berlin hard fork fails with transactions validations for [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559), that was introduced in London hard work, that followed right after Berlin.

▶️  Fixed transactions validations for `EIP-1559`
▶️  Rafactored vicinity transactions validation
▶️  Recaftored `TR_TypeNotSupported` errors

🔐 Now every initial state validation check validation correctness with `expected_error_message` to ensure that validation fail is expected behavior.

### CODE REVIEW

File: evm-tests/jsontests/src/state.rs [hunk 1]
```diff
@@ -722,11 +727,175 @@ fn check_create_exit_reason(
...
+			InvalidTxReason::GasPriseEip1559 => {
+				for (i, state) in states.iter().enumerate() {
+					let expected = state
+						.expect_exception
+						.as_deref()
+						.expect("expected error message for test: [{spec}] {name}:{i}");
+					let is_checked = expected == "TR_TypeNotSupported";
+					assert!(
+						is_checked,
+						"unexpected error message {expected:?} for:  [{spec:?}] {name}:{i}",
```
Comment (aleksuss) [inline evm-tests/jsontests/src/state.rs:749]:
```suggestion
						"unexpected error message {expected:?} for: [{spec:?}] {name}:{i}",
```

File: evm-tests/jsontests/src/state.rs [hunk 2]
```diff
@@ -722,11 +727,175 @@ fn check_create_exit_reason(
...
+					);
+				}
+			}
+			_ => panic!("Unexpected validation reason: {reason:?} [{name}]"),
+		},
+		ForkSpec::Berlin => match reason {
```
Comment (aleksuss) [inline evm-tests/jsontests/src/state.rs:755]: It looks like the branches `ForkSpec::Istanbul` and `ForkSpec::Berlin` could be combined:
```rust
		ForkSpec::Istanbul | ForkSpec::Berlin => match reason {
			InvalidTxReason::GasPriseEip1559 => {
				for (i, state) in states.iter().enumerate() {
					let expected = state
						.expect_exception
						.as_deref()
						.expect("expected error message for test: [{spec}] {name}:{i}");
					let is_checked = expected == "TR_TypeNotSupported";
					assert!(
						is_checked,
						"unexpected error message {expected:?} for:  [{spec:?}] {name}:{i}",
					);
				}
			}
			_ => panic!("Unexpected validation reason: {reason:?} [{name}]"),
		}
```

Comment (mrLSD) [reply evm-tests/jsontests/src/state.rs]: Done: 09cdb6f9634d9b4b76ceaa2f1237c0cecebfbc4d

**Review by aleksuss — APPROVED** [review]

**Review by birchmd — APPROVED** [review]
Nice job!

**Review by aleksuss — APPROVED** [review]

**Pull request merged**